crates/aranya-daemon/src/api.rs

```diff
@@ -351,24 +385,105 @@ impl DaemonApi for Api {
     ) -> api::Result<()> {
         self.check_team_valid(team).await?;
 
-        todo!()
+        match cfg.quic_sync {
+            Some(cfg) => {
+                let psk_store = self
+                    .quic
+                    .as_ref()
+                    .context("quic syncing is not enabled")?
+                    .psk_store
+                    .clone();
+
+                let seed = match cfg.seed_mode {
+                    SeedMode::Generate => {
+                        return Err(api::Error::from_msg(
+                            "Must provide PSK seed from team creation",
+                        ));
+                    }
+                    SeedMode::IKM(ikm) => qs::PskSeed::import_from_ikm(&ikm, team),
+                    SeedMode::Wrapped(wrapped) => {
+                        let enc_sk: EncryptionKey<CS> = {
+                            let enc_id = self.pk.lock().expect("poisoned").enc_pk.id()?;
+                            let crypto = &mut *self.crypto.lock().await;
+                            crypto
+                                .aranya_store
+                                .get_key(&mut crypto.engine, enc_id.into_id())
+                                .context("keystore error")?
+                                .context("missing enc_sk in add_team")?
+                        };
+
+                        let group = GroupId::from(team.into_id());
+                        let seed = enc_sk
+                            .open_psk_seed(
+                                &wrapped.encap_key,
+                                wrapped.encrypted_seed,
+                                &wrapped.sender_pk,
+                                &group,
+                            )
+                            .context("could not open psk seed")?;
+                        qs::PskSeed(seed)
+                    }
+                };
+
+                self.add_seed(team, seed.clone()).await?;
+
+                for psk_res in seed.generate_psks(team) {
+                    let psk = psk_res.context("unable to generate psk")?;
+                    psk_store
+                        .insert(team, Arc::new(psk))
+                        .inspect_err(|err| error!(err = ?err, "unable to insert PSK"))?
+                }
+
+                Ok(())
+            }
+            None => Err(anyhow!("Missing QUIC sync config").into()),
+        }
     }
 
     #[instrument(skip(self))]
     async fn remove_team(self, _: context::Context, team: api::TeamId) -> api::Result<()> {
-        todo!();
+        let mut errors = vec![];
+        if let Some(data) = &self.quic {
+            let _ = data
+                .psk_store
+                .remove(team)
+                .inspect_err(|err| error!(err = ?err, "unable to remove PSK"))
+                .map_err(|err| errors.push(err));
+        }
+
+        let _ = self
+            .seed_id_dir
+            .remove(&team)
+            .await
+            .map_err(|err| errors.push(err));
```

Comment:
If we don't want to return an error if the PSK removal fails here, let's at least log a `warn!`.
```suggestion
        if let Err(e) = self
            .seed_id_dir
            .remove(&team)
            .await
            .map_err(|err| errors.push(err)) {
                warn!(e);
            }
```